lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -2661,75 +2659,207 @@ struct DotOpConversionHelper {
   DotOp dot;
 };
 
-LogicalResult
-DotOpConversion::convertMMA16816(triton::DotOp op, OpAdaptor adapter,
-                                 ConversionPatternRewriter &rewriter) const {
-  Location loc = op->getLoc();
-  MLIRContext *ctx = op->getContext();
-  // D = A * B + C
-  Value A = op.a();
-  Value B = op.b();
-  Value C = op.c();
-  Value D = op.getResult();
-  bool allowTF32 = op.allowTF32();
+// This class helps to adapt the existing DotOpConversion to the latest
+// DotOpOperand layout design. It decouples the exising implementation to two
+// parts:
+// 1. loading the specific operand matrix(for $a, $b, $c) from smem
+// 2. passing the loaded value and perform the mma codegen
+struct MMA16816ConversionHelper {
```

Comment:
Is it necessary to bind the conversion helper with a concrete MMA size? On Ampere, there are three sizes for 1688, 16816 and 1684, considering integer 8 and sparsity, there should be more combinations.

Reply:
Yeah, the original implementations are specific to MMA size because of the different behavior of loading instructions. The migration work also follows the original code to avoid large diversity and keep the debugging work simple.

I think some common abstraction and encapsulation are necessary, and we could continue such work after the mlir-migration is finished and the dot-related benchmarks are ready, with a robust base, the refactoring should be more suitable.